src/frontend/src/components/public/PIASubHeader/index.tsx

```diff
@@ -102,19 +103,36 @@ function PIASubHeader({
     setModalCancelLabel(Messages.GenerateAccessLinkModal.cancelLabel);
   };
   useEffect(() => {
+    // enable finish review button condition
+    // Delegate PIA, MPO_review status, program area at least one role selected and mpo accepted accountability
+    // PI PIA, CPO_review status, same as delegate PIA, also at least one CPO need to accept accountability
     if (
-      pia?.status === PiaStatuses.MPO_REVIEW &&
-      pia?.review?.programArea?.selectedRoles &&
-      pia?.review?.programArea?.selectedRoles?.length > 0 &&
-      pia?.review?.mpo?.isAcknowledged === true &&
-      pia?.review?.mpo?.reviewNote !== ''
+      (pia?.status === PiaStatuses.MPO_REVIEW &&
+        pia?.hasAddedPiToDataElements === false &&
+        pia?.review?.programArea?.selectedRoles &&
+        pia?.review?.programArea?.selectedRoles?.length > 0 &&
+        pia?.review?.mpo?.isAcknowledged === true &&
+        pia?.review?.mpo?.reviewNote !== '') ||
+      (pia?.status === PiaStatuses.CPO_REVIEW &&
+        pia?.hasAddedPiToDataElements !== false &&
+        pia?.review?.programArea?.selectedRoles &&
+        pia?.review?.programArea?.selectedRoles?.length > 0 &&
+        pia?.review?.mpo?.isAcknowledged === true &&
+        pia?.review?.mpo?.reviewNote !== '' &&
+        pia?.review?.cpo &&
+        Object.values(pia?.review.cpo)?.length > 0 &&
+        Object.values(pia?.review?.cpo)?.every(
```

Comment:
Is there no way to reduce this to something less verbose?

Reply:
Maybe split these conditions into variables for readability or write functions.

Reply:
they are all dynamic, do you guys have any ideas?